source/opt/instruction.cpp

```diff
@@ -73,6 +73,7 @@ Instruction::Instruction(IRContext* c, const spv_parsed_instruction_t& inst,
       unique_id_(c->TakeNextUniqueId()),
       dbg_line_insts_(std::move(dbg_line)),
       dbg_scope_(kNoDebugScope, kNoInlinedAt) {
+  operands_.reserve(inst.num_operands);
```

Comment:
Thanks for this optimization!
We were rather lazy in the early days.

Other ideas for future: using a utils::SmallVector instead. But picking a good default size takes care. Also, it currently lacks a reserve method.